sparse/rest/server.go

```diff
@@ -2,15 +2,60 @@ package rest
 
 import (
 	"context"
+	"net"
 	"net/http"
 	"os"
+	"sync"
+	"time"
 
 	"github.com/pkg/errors"
 	log "github.com/sirupsen/logrus"
 
 	"github.com/longhorn/sparse-tools/sparse"
 )
 
+var (
+	defaultIdleTimeout = 90 * time.Second
+)
+
+type IdleTimeer struct {
+	mu          sync.Mutex
+	active      map[net.Conn]bool
+	idleTimeout time.Duration
+	timer       *time.Timer
+}
+
+func NewIdleTimeer(idleTimeout time.Duration) *IdleTimeer {
+	return &IdleTimeer{
+		active:      make(map[net.Conn]bool),
```

Comment:
Suggest adding a RW lock for protecting the map access.

Reply:
ConnState() has `it.lock()` and `defer it.unlock()` in the beginning. Is this what you are referring to?

Reply:
Oh, yeah.